ui/src/message_popup/author/question_page.jsx

```diff
@@ -99,46 +102,34 @@ export default React.createClass({
     this.setState({indicator: _.find(indicators, indicator => indicator.id.toString() === value)});
   },
 
-  onSaveEdit(){
-    console.log("");
-    console.log("Saving the following new question:");
-    const studentIds = this.state.students.map(student => student.id);
-    const text = this.state.questionText;
+  onEditQuestion(){
     const {goodExamples, badExamples} = this.getExamples();
-    const id = _.maxBy(allQuestions, question => question.id).id + 1;
-    const question = {
-      studentIds,
-      id,
-      text,
+    const {originalQuestion} = this.props;
+    const newQuestion = {
+      studentIds: this.state.students.map(student => student.id),
+      text: this.state.questionText,
       examples: goodExamples,
       nonExamples: badExamples,
       indicatorId: this.state.indicator.id,
-      indicator: this.state.indicator,
+      indicator: this.state.indicator
     };
-    console.log(question);
-    console.log("Also deleting and archiving the following question:");
-    console.log(this.props.originalQuestion);
-    console.log("");
+    if(this.props.onEditQuestion === undefined) return;
```

Comment:
I tried returning at the beginning of the method if it was not defined but flow wouldn't recognize it unless it was right before when the method was called.